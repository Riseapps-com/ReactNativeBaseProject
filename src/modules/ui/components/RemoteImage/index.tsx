import React from 'react';
import { Image } from 'react-native';

import type { ImageProps, ImageStyle, StyleProp } from 'react-native';

type RemoteImageProps = Omit<ImageProps, 'source'> & {
  width?: number;
  height?: number;
  source: string;
};

const RemoteImage: React.FC<RemoteImageProps> = props => {
  const { width, height, source, style, ...restProps } = props;
  const imageStyles: StyleProp<ImageStyle> = [!!width && { width }, !!height && { height }, style];

  return <Image {...restProps} accessibilityLabel={source} source={{ uri: source }} style={imageStyles} />;
};

export default RemoteImage;
