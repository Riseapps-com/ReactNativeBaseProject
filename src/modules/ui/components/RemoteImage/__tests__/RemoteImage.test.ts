import { renderComponent } from '~modules/tests';

import RemoteImage from '../index';

import type { RemoteImageProps } from '../index';

const renderRemoteImage = (props: RemoteImageProps) => renderComponent(RemoteImage, props);

describe('ui', () => {
  describe('<RemoteImage />', () => {
    it('renders <LocalImage />', () => {
      const remoteImage = renderRemoteImage({
        source: 'https://testImage.com',
      });

      expect(remoteImage.getByA11yLabel('https://testImage.com')).toBeTruthy();
    });

    it('renders custom width and height', () => {
      const remoteImage = renderRemoteImage({
        source: 'https://testImage.com',
        width: 100,
        height: 200,
      });

      expect(remoteImage.getByA11yLabel('https://testImage.com')).toHaveStyle({ width: 100, height: 200 });
    });
  });
});
