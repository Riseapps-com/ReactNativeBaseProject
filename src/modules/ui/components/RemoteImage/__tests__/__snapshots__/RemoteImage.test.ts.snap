// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <RemoteImage /> renders <LocalImage /> 1`] = `
<Image
  accessibilityLabel="https://testImage.com"
  source={
    Object {
      "uri": "https://testImage.com",
    }
  }
  style={
    Array [
      false,
      false,
      undefined,
    ]
  }
/>
`;
